### PR TITLE
[feature/APPC-1037] - Implement empty transactions screen carousel

### PR DESCRIPTION
**Pull request by joaopmgouveia**
**What does this PR do?**

   Adds a ViewPager with 2 pages, one for the old empty transactions screen and one for the new carousel. Now both the clickable text and animation are clickable: the old screen still opens the gamification screen and the carousel opens either the Aptoide Appstore app or the browser

**Database changed?**

   No

**Where should the reviewer start?**

- EmptyTransactionsView.java

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APPC-1037](https://aptoide.atlassian.net/browse/APPC-1037)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1037](https://aptoide.atlassian.net/browse/APPC-1037)

**Questions:**

   - Infinite swipe implementation was taking too much time and thus is not included in this PR

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/widget/EmptyTransactionsView.java [hunk 1]
```diff
@@ -3,32 +3,65 @@
...
+  private final int action[] =
+      { R.string.home_empty_discover_apps_button, R.string.gamification_home_button };
 
-  public EmptyTransactionsView(@NonNull Context context, OnClickListener onClickListener,
-      @NonNull String bonus) {
+  public EmptyTransactionsView(@NonNull Context context, @NonNull String bonus,
+      PublishSubject<String> emptyTransactionsSubject, TransactionsActivity transactionsActivity) {
     super(context);
 
+    disposables = new CompositeDisposable();
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/widget/EmptyTransactionsView.java]: Initialise it in the activity and pass it in the constructor

File: app/src/main/java/com/asfoundation/wallet/widget/EmptyTransactionsView.java [hunk 2]
```diff
@@ -3,32 +3,65 @@
...
-  private final TextView noTransactionsTextView;
-  private final TextView earnBonusTextView;
-  private final LottieAnimationView noTransactionsAnimationView;
+  private static final int MAX_BONUS_STRING_RESOURCE = R.string.gamification_home_body;
+  private static final int NUMBER_PAGES = 2;
+  private final CompositeDisposable disposables;
+  private final int anim[] =
+      { R.raw.carousel_empty_screen_animation, R.raw.transactions_empty_screen_animation };
+  private final int body[] = { R.string.home_empty_discover_apps_body, MAX_BONUS_STRING_RESOURCE };
+  private final int action[] =
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/widget/EmptyTransactionsView.java]: These array can be created locally

File: app/src/main/java/com/asfoundation/wallet/ui/TransactionsActivity.java [hunk 3]
```diff
@@ -154,7 +159,8 @@ public static Intent newIntent(Context context) {
 
   private void onFetchTransactionsError(Double maxBonus) {
     if (emptyView == null) {
-      emptyView = new EmptyTransactionsView(this, this, String.valueOf(maxBonus));
+      emptyView =
+          new EmptyTransactionsView(this, String.valueOf(maxBonus), emptyTransactionsSubject, this);
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/ui/TransactionsActivity.java]: I’m not sure, but I think that if you move the systemView.showEmpty outside of the if, it should solve the problem of launching more than 1 intent

File: app/src/main/java/com/asfoundation/wallet/ui/TransactionsActivity.java [hunk 4]
```diff
@@ -278,7 +280,8 @@ private void onDefaultNetwork(NetworkInfo networkInfo) {
   private void onError(ErrorEnvelope errorEnvelope) {
     if ((errorEnvelope.code == EMPTY_COLLECTION || adapter.getItemCount() == 0)) {
       if (emptyView == null) {
-        emptyView = new EmptyTransactionsView(this, this, String.valueOf(maxBonusEmptyScreen));
+        emptyView = new EmptyTransactionsView(this, String.valueOf(maxBonusEmptyScreen),
+            emptyTransactionsSubject, this);
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/ui/TransactionsActivity.java]: Same as above

File: app/src/main/java/com/asfoundation/wallet/ui/widget/adapter/EmptyTransactionPagerAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.viewpager.widget.PagerAdapter
+import androidx.viewpager.widget.ViewPager
+import com.airbnb.lottie.LottieAnimationView
+import com.asf.wallet.R
+import io.reactivex.subjects.PublishSubject
+import org.jetbrains.annotations.NotNull
+
+class EmptyTransactionPagerAdapter(private val anim: IntArray,
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/ui/widget/adapter/EmptyTransactionPagerAdapter.kt]: variable name: anim -> animation

**Review by JDSM01 — CHANGES_REQUESTED** [review]
1- When you scroll to the opposite side of the other view, the animation restarts.
2- When you click on “Discover Top Apps” or its image and you cancel in the complete action dialog and then you click on the text or view again it launches 2 dialogs instead of 1. Every time you do this it adds 1 more dialog.
3- By solving number 2, this will probably be solved. the randomize should only not occur when on resume (when the dialog opens and you cancel it)

**Review by carvalho2707 — APPROVED** [review]

**Pull request merged**